go/chat/server.go

```diff
@@ -460,206 +462,25 @@ func (h *Server) GetThreadNonblock(ctx context.Context, arg chat1.GetThreadNonbl
 
 // NewConversationLocal implements keybase.chatLocal.newConversationLocal protocol.
 // Create a new conversation. Or in the case of CHAT, create-or-get a conversation.
-func (h *Server) NewConversationLocal(ctx context.Context, arg chat1.NewConversationLocalArg) (res chat1.NewConversationLocalRes, reserr error) {
+func (h *Server) NewConversationLocal(ctx context.Context, arg chat1.NewConversationLocalArg) (res chat1.NewConversationLocalRes, err error) {
 	var identBreaks []keybase1.TLFIdentifyFailure
 	ctx = Context(ctx, h.G(), arg.IdentifyBehavior, &identBreaks, h.identNotifier)
-	defer h.Trace(ctx, func() error { return reserr }, "NewConversationLocal")()
+	defer h.Trace(ctx, func() error { return err }, "NewConversationLocal")()
 	if err := h.assertLoggedIn(ctx); err != nil {
 		return chat1.NewConversationLocalRes{}, err
 	}
 
-	// Handle a nil topic name with default values for the members type specified
-	if arg.TopicName == nil {
-		// We never want a blank topic name in team chats, always default to the default team name
-		switch arg.MembersType {
-		case chat1.ConversationMembersType_TEAM:
-			arg.TopicName = &DefaultTeamTopic
-		}
-	}
-
-	// Find any existing conversations that match this argument specifically. We need to do this check
-	// here in the client since we can't see the topic name on the server.
-
-	// NOTE: The CLI already does this. It is hard to move that code completely into the service, since
-	// there is a ton of logic in there to try and present a nice looking menu to help out the
-	// user and such. For the most part, the CLI just uses FindConversationsLocal though, so it
-	// should hopefully just result in a bunch of cache hits on the second invocation.
-	findArg := chat1.FindConversationsLocalArg{
-		TlfName:          arg.TlfName,
-		MembersType:      arg.MembersType,
-		Visibility:       arg.TlfVisibility,
-		TopicType:        arg.TopicType,
-		IdentifyBehavior: arg.IdentifyBehavior,
-	}
-	if arg.TopicName != nil {
-		findArg.TopicName = *arg.TopicName
-	}
-	findRes, err := h.FindConversationsLocal(ctx, findArg)
-	if err != nil {
-		return chat1.NewConversationLocalRes{}, err
-	}
-	// If we find one conversation, then just return it as if we created it.
-	if len(findRes.Conversations) == 1 {
-		return chat1.NewConversationLocalRes{
-			Conv:             findRes.Conversations[0],
-			RateLimits:       findRes.RateLimits,
-			IdentifyFailures: findRes.IdentifyFailures,
-		}, nil
-	}
-	res.RateLimits = append(res.RateLimits, findRes.RateLimits...)
-
-	info, err := CtxKeyFinder(ctx, h.G()).Find(ctx, arg.TlfName, arg.MembersType,
-		arg.TlfVisibility == chat1.TLFVisibility_PUBLIC)
+	uid := gregor1.UID(h.G().Env.GetUID().ToBytes())
```

Comment:
Yeah I just don't like to handle the error there, since right above this I just called `assertLoggedIn`, so if we make it this far I should just be able to pull the uid without having to add the verbiage of the error check.